cockatrice/src/player.cpp

```diff
@@ -1095,20 +1095,34 @@ void Player::actCreatePredefinedToken()
 
 void Player::actCreateRelatedCard()
 {
-    // get the clicked card
     CardItem * sourceCard = game->getActiveCard();
     if(!sourceCard)
         return;
 
-    // get the target card name
     QAction *action = static_cast<QAction *>(sender());
+    const QString &actionDisplayName = action->text();
+    createCard(sourceCard, dbNameFromTokenDisplayName(actionDisplayName));
+}
+
+void Player::actCreateAllRelatedCards()
+{
+    CardItem * sourceCard = game->getActiveCard();
+    if(!sourceCard)
+        return;
+
+    QStringList relatedCards = * new QStringList();
+    relatedCards.append(sourceCard->getInfo()->getRelatedCards());
+    relatedCards.append(sourceCard->getInfo()->getReverseRelatedCards2Me());
```

Comment:
While creating a new QStringList makes the code more readable, the second "append" forces Qt to make a deep copy of the list, with all the overhead involved in memory allocation, copy, etc..
This is not a big deal since the "related cards" list would only contain a few hundreds of bytes at max..